### ISSUE TITLE
howto convert enums to union in [jsonschema] [typescript]

### ISSUE THREAD

**Issue by NewFuture** [glideapps/quicktype#1234]
```json
{
 "data_type": {
            "type": "string",
            "title": "DataType",
            "description": "Medium data type (link raw or base64)",
            "examples": [
              "raw"
            ],
            "enum": [
              "link",
              "raw",
              "base64"
            ]
      }
}
```
whem convert with ` --just-types` (as `.d.ts`)

```ts
export enum DataType {
    Base64 = "base64",
    Link = "link",
    Raw = "raw",
}
```

it's better to be union
```ts
export type DataType  = "base64"|"link"|"raw";
```

**Comment by altruisticsoftware** [comment]
Would really like to have this.

**Comment by cmawhorter** [comment]
i also would like to have this. it makes consuming the generated types very painful. 

especially because of the lack of ability to [prefix the generated enum](https://github.com/quicktype/quicktype/issues/1372) which means exposing those enums is more trouble than its worth and it's easier just to do `'base64' as any` ... which is not ideal.

**Comment by cmawhorter** [comment]
this finally got me to dig into the quicktype code and was pleasantly surprised that this is actually really easy to fix programatically.

you can extend the default language class and override the enum behavior.  this should be copy/pastable:

```ts
import { quicktype, InputData, JSONSchemaInput, JSONSchemaStore, TypeScriptTargetLanguage, TypeScriptRenderer, RenderContext, getOptionValues, tsFlowOptions, EnumType, Name, TargetLanguage, SerializedRenderResult } from 'quicktype-core';
import { utf16StringEscape } from 'quicktype-core/dist/support/Strings';

export class MyTypeScriptTargetLanguage extends TypeScriptTargetLanguage {
  protected makeRenderer(
    renderContext: RenderContext,
    untypedOptionValues: { [name: string]: any }
  ): MyTypeScriptRenderer {
      return new MyTypeScriptRenderer(this, renderContext, getOptionValues(tsFlowOptions, untypedOptionValues));
  }
}

export class MyTypeScriptRenderer extends TypeScriptRenderer {
  protected emitEnum(e: EnumType, enumName: Name): void {
    this.emitDescription(this.descriptionForType(e));
    this.emitLine(["export type ", enumName, " = "]);
    this.forEachEnumCase(e, "none", (name, jsonName, position) => {
      const suffix = position === 'last' || position === 'only' ? ';' : ' | ';
      this.indent(() =>
        this.emitLine(`"${utf16StringEscape(jsonName)}"`, suffix));
    });
  }
}
```

and passing that custom language to quicktype:

```ts
await quicktype({
    inputData,
    lang: new MyTypeScriptTargetLanguage(),
})
```

the result is:

```ts
export interface SomeClass {
  someEnum: SomeEnum;
}

export type SomeEnum = 
  "one" |
  "two" |
  "three";
```

which is good enough for me.

**Comment by dangdennis** [comment]
@cmawhorter please let me buy you coffee or your favorite beverage. this was a lifesaver.

**Comment by cmawhorter** [comment]
Ha, @dangdennis. Np. Glad it was useful.

**Comment by DavidTanner** [comment]
This should be closed by #1908
